extensions-contrib/kafka-eight-simpleConsumer/src/main/java/org/apache/druid/firehose/kafka/KafkaSimpleConsumer.java

```diff
@@ -274,7 +272,7 @@ private long getOffset(boolean earliest) throws InterruptedException
       }
     }
 
-    return response != null ? filterAndDecode(response.messageSet(topic, partitionId), offset) : EMPTY_MSGS;
+    return filterAndDecode(response.messageSet(topic, partitionId), offset);
```

Comment:
Is it impossible for `response` to be null at this point? I guess the idea is that if it's null, `errorCode` should be `ErrorMapping.UnknownCode()`, and therefore `needNewLeader` should be true, and so we should hit the `continue;` and keep looping.

The logic is a bit nontrivial to follow, so if that is accurate, a comment or `Preconditions.checkState` would be helpful.

Reply:
No, it's impossible. Open this class in IntelliJ - its data flow analysis identifies that the condition is always true as well.

Related to #7780.